src/lib/CPAuth/CPAuth.ts

```diff
@@ -0,0 +1,16 @@
+import { defaultConfig } from 'lib/CPAuth/config';
+import OAuth2 from 'lib/OAuth2/OAuth2';
+
+class CPAuth extends OAuth2 {
```

Comment:
This inherits from the OAuth2 class, so we could have this as a singleton, but not limit the number of OAuth2 providers to a single one.